### PR TITLE
Bug 1169286 - Set anchor for context menu popup

### PR DESCRIPTION
**Pull request by thebnich**

### CODE REVIEW

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 1]
```diff
@@ -1742,6 +1742,13 @@ extension BrowserViewController: ContextMenuHelperDelegate {
             actionSheetController.addAction(copyAction)
         }
 
+        // If we're showing an arrow popup, set the anchor to the long press location.
+        if let popoverPresentationController = actionSheetController.popoverPresentationController {
+            popoverPresentationController.sourceView = view
+            popoverPresentationController.sourceRect = CGRect(origin: gestureRecognizer.locationInView(view), size: CGSizeMake(0, 16))
```
Comment (st3fan) [inline Client/Frontend/Browser/BrowserViewController.swift]: To be closer to the touch point I would do `CGRectInset(origin: ..., size: CGSizeZero), -8, -8)` .. then the rect is around the touch point.

File: Client/Frontend/Browser/ContextMenuHelper.swift [hunk 2]
```diff
@@ -31,9 +32,8 @@ class ContextMenuHelper: NSObject, BrowserHelper, UIGestureRecognizerDelegate {
         browser.webView.configuration.userContentController.addUserScript(userScript)
 
         // Add a gesture recognizer that disables the built-in context menu gesture recognizer.
-        let disableRecognizer = UILongPressGestureRecognizer()
-        disableRecognizer.delegate = self
-        browser.webView.addGestureRecognizer(disableRecognizer)
+        gestureRecognizer.delegate = self
```
Comment (st3fan) [inline Client/Frontend/Browser/ContextMenuHelper.swift]: This creates a reference loop. In the docs, the `delegate` is marked as `unowned(unsafe)` .. is that similar to weak?

Comment (thebnich) [inline Client/Frontend/Browser/ContextMenuHelper.swift]: Yeah, it's basically an implicitly unwrapped `weak`.

File: Client/Frontend/Browser/BrowserViewController.swift (hunk 1 above)
Comment (thebnich) [inline Client/Frontend/Browser/BrowserViewController.swift]: Hm, the location looks correct to me when I test in the simulator. Where are you getting (-8, -8) from?

Comment (thebnich) [inline Client/Frontend/Browser/BrowserViewController.swift]: Looks OK to me in the iPad simulator, but please file a follow-up if you see any issues!

**Pull request merged**